waddrmgr/db.go

```diff
@@ -1400,3 +1738,140 @@ func upgradeManager(namespace walletdb.Namespace) error {
 
 	return nil
 }
+
+// upgradeToVersion3 upgrades the database from version 2 to version 3
+// The following buckets were introduced in version 3 to support account names:
+// * acctNameIdxBucketName
+// * acctIdIdxBucketName
+// * metaBucketName
+func upgradeToVersion3(namespace walletdb.Namespace) error {
+	err := namespace.Update(func(tx walletdb.Tx) error {
+		currentMgrVersion := uint32(3)
+		rootBucket := tx.RootBucket()
+
+		reader := bufio.NewReader(os.Stdin)
+
+		seed, err := cmd.PromptExistingConsoleSeed(reader)
```

Comment:
The upgrade function must not prompt the user directly. This information needs to be provided by the caller through some kind of callback mechanism.

Keep in mind that the waddrmgr is intended to be useful for more than just wallet, and ultimately will be used in a GUI too.